### DISCUSSION TITLE
a problem with subscription changes on log.

### DISCUSSION THREAD

**Discussion by amirmahdifard** [BearWare/TeamTalk5#1524] [Q&A]
### Description

hi. just updated my server to 5.11  and tested the subscription changes in the server. onistly, after each subscription, theres bunch of leters, but the problem is, there always difrent and as much as i read them, i really can't understand it's wich subscription, and it's turned off or on. please chek and remove the leters that we don't  need, or if we  need them, please explane to me. i understand some of them, but they really useless. for example, when i intercept only my voice, and it says um,cm,bm,mm,mm or something like that. so please dabble chek. thank you.

### Application

- [ ] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [X] TeamTalkServer

**Comment by bear101** [comment]
You can see how the subscriptions are converted into text here: https://github.com/BearWare/TeamTalk5/blob/master/Library/TeamTalkLib/bin/ttsrv/ServerGuard.cpp#L278

**Comment by amirmahdifard** [comment]
but i'm not understanding this. please explane to me in a short way. thank you.

**Comment by lbk2907** [comment]
um: User messages
cm: Channel messages.
bm: Broadcast messages.
mm: I not really sure what is it.
vo: Voice.
vi: Video.
de: Desktop.
di: Desktop access.
mf: Media file.
Any letter that starts with capital letter is the intercept subscription.
@bearware101
Correct me if I wrong.

**Reply by bear101** [reply]
Correct!

**Reply by bear101** [reply]
"mm" is custom message. It's e.g. used to indicate a user is typing text in a text dialog

**Comment by lbk2907** [comment]
@bearware101
May I know what is for mm?

**Comment by lbk2907** [comment]
@bearware101
I'm wondering how TeamTalk client can subscribe to custom messages. Are
possible?

**Reply by bear101** [reply]
It's not possible in the TeamTalk client to unsubscribe custom messages.

**Comment by lbk2907** [comment]
@bear101
Thanks

**Comment by amirmahdifard** [comment]
i understand this, but how to understand that for example, someone turned  my subscription on or off?

**Comment by lbk2907** [comment]
It doesn't show up in the log if the particular event has not be subscribe.

**Comment by amirmahdifard** [comment]
but please make it so we can understand it's turned off or on. thanks.
